File: algo.cpp
#include<bits/stdc++.h>
using namespace std;
#define int long long

int grid[105][105];
pair<int,int> points[10100];
int lines [10100][3];



int vis[105][105]={0};
int c=0;
void dfs(int x,int y)
{
    if(vis[y][x]==0 && grid[y][x]==0)
    {
        vis[y][x]=1;
        points[c].first=y;
        points[c].second=x;
        c++;
        dfs(x+1,y);
        dfs(x-1,y);
        dfs(x,y+1);
        dfs(x,y-1); 
    }
}

int32_t main()
{
    ios::sync_with_stdio(false);
    cin.tie(0);
    memset(grid,-1,sizeof(grid));
    int n,m;
    cout<<"Enter grid size";
    cin>>n>>m;//size of grid
    //setting lines at boundry
    for(int i=0;i<=n+1;i++)
        grid[i][0]=1,grid[i][m+1]=1;
    
    for(int j=0;j<=m+1;j++)
        grid[0][j]=1,grid[n+1][j]=1;
    cout<<"Enter Grid";
    for(int i=1;i<=n;i++)
        for(int j=1;j<=m;j++)
            cin>>grid[i][j];// 0 for non lines points// 1 for lines points
    
    int x;
    int y;
    cout<<"Enter point\n";
    cin>>x>>y;//point to be colored
    dfs(x,y);
    sort(points,points + c);
    int l=0;
    int empty=0;
    int left,right,height;
    for(int i=0;i<c;i++)
    {
        if(empty==0)
        {
            empty=1;
            height=points[i].first;
            left=points[i].second;
            right=points[i].second;
        }
        else if(height != points[i].first || right+1 != points[i].second)//lines has been broken
        {
            lines[l][0]=height;
            lines[l][1]=left;
            lines[l][2]=right;
            l++;
            height=points[i].first;
            left=points[i].second;
            right=points[i].second;
            
        }
        else
            right++;
        if(i==c-1)
        {
            lines[l][0]=height;
            lines[l][1]=left;
            lines[l][2]=right;
            l++;
        }
    }
    for(int i=0;i<l;i++)
        cout<<lines[i][0]<<" "<<lines[i][1]<<" "<<lines[i][2]<<"\n";// in the order y-cordinate then left right
}

File: algo.py
class Scan():
    def __init__(self,origin,grid):    #### Initializes all arrays and variables of class
        self.origin = origin
        self.grid = grid
        self.set_grid = []
        self.points = []
        self.visited = []
        self.lines = []
        self.count = 0

    def initializeArrays(self):                      ###Initializes the arrays according to the grid on screen 
        for j in range(len(self.grid.getGrid())):
            self.set_grid.append([])
            self.visited.append([])
            for i in range(len(self.grid.getGrid()[j])):
                curr_pt = self.grid.getGrid()[i][j]
                if (curr_pt.getColor() == (0,0,0)):
                    self.set_grid[j].append(1)
                else:
                    self.set_grid[j].append(0)
                self.visited[j].append(0)

    def dfs(self,x,y):           ### DFS flood fill to detect all point to be coloured
        ## x - column
        ## y - row
        if (0 <= x < len(self.set_grid) and 0 <= y < len(self.set_grid[0]) and self.visited[x][y] == 0 and self.set_grid[y][x] == 0):
            self.visited[x][y] = 1
            self.points.append([y,x])
            self.count+=1
            self.dfs(x+1,y)
            self.dfs(x-1,y)
            self.dfs(x,y+1)
            self.dfs(x,y-1)

    def Sort(self,sub_li,index):          ### Sorts the given array according the to given index of sub-array
    # reverse = None (Sorts in Ascending order) 
    # key is set to sort using second element of  
    # sublist lambda has been used 
        sub_li.sort(key = lambda x: x[index]) 
        return sub_li 

    def main(self):                       ### Main function to return list of lines to be coloured sorted by height and the left and right extreme coordinates
        self.initializeArrays()
        self.dfs(self.origin.col,self.origin.row)
        self.points = self.Sort(self.points,1)
        self.points = self.Sort(self.points,0)
        l,right,left,height = 0,0,0,0
        first_pt = True
        for i in range(self.count):
            if (first_pt):
                first_pt = False
                height = self.points[i][0]
                left = self.points[i][1]
                right = self.points[i][1]
            elif (height != self.points[i][0] or right+1 != self.points[i][1]):
                self.lines.append([height,left,right])
                height = self.points[i][0]
                left = self.points[i][1]
                right = self.points[i][1]
            else:
                right+=1

            if (i == self.count-1):
                self.lines.append([height,left,right])
        
        # for x in self.lines:
        #     print (x)
        # print ('\n')

        return self.lines


